src/TimeoutMigrationTool.ASQ.IntegrationTests/DelayedDeliveryTableNameProviderTests.cs

```diff
@@ -0,0 +1,39 @@
+﻿namespace TimeoutMigrationTool.ASQ.IntegrationTests
+{
+    using NUnit.Framework;
+    using Particular.TimeoutMigrationTool.ASQ;
+    using System;
+    using System.Collections.Generic;
+    using System.Text;
+    using System.Threading.Tasks;
+
+    [TestFixture]
+    class DelayedDeliveryTableNameProviderTests
+    {
+        [Test]
+        public void When_Migrating_Endpoint_With_No_DelayedDeliveryTableOverride_DelayedDeliveryTableName_Is_Computed()
+        {
+            // Arrange
+            var delayedDeliveryTableNameGenerator = new DelayedDeliveryTableNameProvider(null);
+
+            // Act
+            var delayedDeliveryTableName = delayedDeliveryTableNameGenerator.GetDelayedDeliveryTableName("EndpointName");
+
+            // Assert
+            Assert.IsTrue(delayedDeliveryTableName.StartsWith("delays"));
+        }
+
+        [Test]
+        public void When_Migrating_Endpoint_With_DelayedDeliveryTableOverride_DelayedDeliveryTableName_Is_NotComputed()
+        {
+            // Arrange
+            var delayedDeliveryTableNameGenerator = new DelayedDeliveryTableNameProvider("overriden delayed delivery table name");
+
+            // Act
+            var delayedDeliveryTableName = delayedDeliveryTableNameGenerator.GetDelayedDeliveryTableName("EndpointName");
+
+            // Assert
+            Assert.AreEqual("overriden delayed delivery table name", delayedDeliveryTableName);
```

Comment:
That doesn't seem to be a valid azure table name, is that OK?

Reply:
This is a unit test, not touching azure anywhere so I'm not sure it's necessary to be a valid azure table name